src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -299,8 +300,12 @@ private DiagnosticMode GetDiagnosticMode(RequestContext context)
 
             context.TraceInformation($"Found {diagnostics.Length} diagnostics for {diagnosticSource.ToDisplayString()}");
 
+            var document = diagnosticSource.GetTextDocument();
+            var sourceText = document is null ? null : await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
to align with the roslyn squiggle-tagger, we attempt to ensure that the spans we're dealing with are appropriately placed on the source-text we have.